### ISSUE TITLE
[BUG] Markers on ribbon inconsistent

### ISSUE THREAD

**Issue by eddieschoute** [JuliaPlots/Plots.jl#3537]
## Details
I want to have a ribbon with markers on the datapoints, but not the ribbon.
For example,

```
using Plots

pgfplotsx()
plot(
    1:10,
    rand(10),
    ribbon = 0.1 * rand(1:10),
    markers = :auto
)
savefig("pgfplot.png")

gr()
plot(
    1:10,
    rand(10),
    ribbon = 0.1 * rand(1:10),
    markers = :auto
)
savefig("grplot.png")
```

Gives
![pgfplot](https://user-images.githubusercontent.com/25747714/120658419-6e7dcd00-c453-11eb-95fa-bd7ce94bb781.png)
![grplot](https://user-images.githubusercontent.com/25747714/120658440-70e02700-c453-11eb-8d4e-fa588b686c7a.png)
for pgf, gr respectively.
I want the GR behavior because I think markers on a ribbon do not make sense and it add visual clutter.

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |     |  x   |
pyplot       |     |   x  |
plotly       |  x   |     |
plotlyjs     |    |     | x
pgfplotsx    |  x   |     |
inspectdr    |     |     | x

### Versions

Plots.jl version: 1.15.3
Backend  version (`]st -m`): pgfplotsx 1.2.10, gr 0.57.5
Output of `versioninfo()`: 1.6.1

**Comment by BeastyBlacksmith** [comment]
meanwhile you can workaround by adding the markers after the fact
```julia
using Plots

pgfplotsx()
xs = 1:10
ys = rand(10)
ribs = 0.1 * rand(1:10)
plot(
    xs,
    ys,
    ribbon = ribs,
    markers = :none
)
scatter!( xs, ys, seriescolor = 1 )

savefig("pgfplot.png")
```

**Comment by eddieschoute** [comment]
Unfortunately, if primary=false then the `palette` attribute seems to be ignored, resulting in the same color for all curves.

**Comment by BeastyBlacksmith** [comment]
you got an MWE for what you want to do?

**Comment by eddieschoute** [comment]
Here is an example
```
palette_main = palette(:Paired_4, 4)[[1,3]]

markers = reshape([
    :circle
    :utriangle
    :rect
    :dtriangle
], 1, :)
y = [rand(10) rand(10).+0.5 rand(10).+2 rand(10).+2.5]
pgfplotsx()
plot(
    1:10,
    y,
    ribbon = 0.1 * rand(1:10),
    markershape = markers,
    palette = palette_main
)
```
![pgfplot](https://user-images.githubusercontent.com/25747714/121039739-5a9dd800-c77f-11eb-8332-a02a295ef356.png)

I would want no markers on the ribbon but I do want markers in the legend. There are sets (here of size 2) of curves of different colors at different levels of y.

**Comment by BeastyBlacksmith** [comment]
The best I can do at the moment is
```julia
plot(
           1:10,
           y,
           ribbon = 0.1 * rand(1:10),
           palette = palette_main, label = false
       )
scatter!(y, marker = markers, palette = palette_main)
```

**Comment by eddieschoute** [comment]
Yes that's really close to what I'd want. I think it'll do for now, thanks!

**Comment by eddieschoute** [comment]
Thanks @BeastyBlacksmith , have you looked at the legend? It seems the markers for the legend still get lost in PGFPlotsX
